File: Leetcode/1317 - Convert Integer to the Sum of Two No-Zero Integers/solution.cpp
bool isNoZero(int n) {
    while (n) {
        if (n % 10 == 0) return false;
        n /= 10;
    }
    return true;
}

vector<int> getNoZeroIntegers(int n) {
    for (int i = 1; i <= n / 2; i++) {
        if (isNoZero(i) && isNoZero(n-i)) {
            return {i, n-i};
        }
    }
    return {};
}

File: Leetcode/1317 - Convert Integer to the Sum of Two No-Zero Integers/solution.py
def getNoZeroIntegers(n: int) -> List[int]:
    def is_no_zero(num: int) -> bool:
        while num > 0:
            if num % 10 == 0:
                return False
            num //= 10
        return True

    for i in range(1, n // 2 + 1):
        if is_no_zero(i) and is_no_zero(n - i):
            return [i, n - i]
    return []